### PR TITLE
Enhancement: Implement Test Cases

### PR DESCRIPTION
**Pull request by ahmidou09**
### Description
This pull request addresses the following tasks:

- Added comprehensive unit tests for all methods in the Book, Person, Student, Teacher, Classroom, and Rental classes.
- Ensured that all tests pass and no linter errors are present.

### Changes Made
- Implemented test cases for each method in the specified classes.
- Verified that each method behaves as expected and edge cases are handled.

### CODE REVIEW

File: spec/person_spec.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
+require 'date'
+require_relative '../person/person'
+
+describe Person do
```
Comment (zeff96) [inline spec/person_spec.rb:4]: - [x] Hello! 🌟 Please ensure that you include a test case for the correct_name method within the Person class. By doing so, you'll guarantee that all implemented methods are thoroughly tested, aligning perfectly with the project guidelines. 🧪 This step is essential to validate the correctness and robustness of your code. 🚀

File: spec/teacher_spec.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
+require 'date'
+require_relative '../person/teacher'
+
+describe Teacher do
```
Comment (zeff96) [inline spec/teacher_spec.rb:4]: - [x] Greetings! 🌟 It's important to bolster test coverage for all methods present in the Teacher class, especially the to_s method, which is currently untested. By addressing this, you will be in perfect alignment with the project's requirements. 🧪 This step is crucial to validate the effectiveness and completeness of your code. 🚀

File: spec/student_spec.rb [hunk 3]
```diff
@@ -0,0 +1,34 @@
+require 'date'
+require_relative '../person/student'
+require_relative '../person/classroom'
+
+describe Student do
```
Comment (zeff96) [inline spec/student_spec.rb:5]: - [x] Greetings! 🌟 It's important to bolster test coverage for all methods present in the Student class, especially the to_s method, which is currently untested. By addressing this, you will be in perfect alignment with the project's requirements. 🧪 This step is crucial to validate the effectiveness and completeness of your code. 🚀

**Review by zeff96 — CHANGES_REQUESTED** [review]
### Status: CHANGES REQUIRED ♻️
Hi @ahmidou09, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost

## Highlights
:heavy_check_mark: Linters passing
:heavy_check_mark: Good commit messages
:heavy_check_mark: Correct GitFlow

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ahmidou09** [comment]
Thank you for your review @zeff96

**Review by shahadat3669 — APPROVED** [review]
Hi @ahmidou09, 

Your project is complete!  There is nothing else to say other than it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
✅ All test files should be placed in a spec directory
✅ There should be one file per class
✅ there is an 'it' block for each class method (at least)

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

 

Cheers and Happy coding!👏👏👏

## Have any doubt ❓

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

<table  width="100%" align="center">
  <tr>
    <td colspan="4" align="center">
      <strong>Contact me: Shahadat Hossain</strong>
    </td>
  </tr>
  <tr>
    <td align="center">
      <a href="https://github.com/shahadata3669">
        GitHub
      </a>
    </td>
    <td align="center">
      <a href="https://www.linkedin.com/in/shahadat-cseng/">
        LinkedIn
      </a>
    </td>
    <td align="center">
      <a href="https://app.slack.com/client/T47CT8XPG/D05GT4E4WAY/rimeto_profile/U04426SD71V">
        Slack
      </a>
    </td>
    <td align="center">
      <a href="mailto:shahadat3669@gmail.com?subject=Microverse%20Code%20Review">
        Email
      </a>
    </td>
  </tr>
</table>

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ahmidou09** [comment]
Thank you for your review @shahadat3669
